packages/core-data/src/private-selectors.ts

```diff
@@ -50,3 +51,24 @@ export const getBlockPatternsForPostType = createRegistrySelector(
 			() => [ select( STORE_NAME ).getBlockPatterns() ]
 		)
 );
+
+/**
+ * Returns the entity records permissions for the given entity record ids.
+ */
+export const getEntityRecordsPermissions = createSelector(
+	( state: State, kind: string, name: string, ids: string[] ) => {
+		return ids.map( ( id ) => ( {
+			delete: canUser( state, 'delete', {
```

Comment:
We now have two canUser calls, but I checked and we still make one request per id and not two so it is fine.

Reply:
Yes, requests are deduped for related actions - #43480.